syft/pkg/cataloger/dotnet/config.go

```diff
@@ -0,0 +1,255 @@
+type nugetProviderCredential struct {
+	Username string `yaml:"username" json:"username" mapstructure:"username"`
+	Password string `yaml:"password" json:"password" mapstructure:"password"`
+}
+
+type CatalogerConfig struct {
+	SearchLocalLicenses  bool                      `yaml:"search-local-licenses" json:"search-local-licenses" mapstructure:"search-local-licenses"`
+	SearchRemoteLicenses bool                      `yaml:"search-remote-licenses" json:"search-remote-licenses" mapstructure:"search-remote-licenses"`
+	Providers            []string                  `yaml:"package-providers,omitempty" json:"package-providers,omitempty" mapstructure:"package-providers"`
+	ProviderCredentials  []nugetProviderCredential `yaml:"package-provider-credentials,omitempty" json:"package-provider-credentials,omitempty" mapstructure:"package-provider-credentials"`
+}
+
+// DefaultCatalogerConfig create a CatalogerConfig with default options, which includes:
+// - setting the default remote package providers if none are provided
+func DefaultCatalogerConfig() CatalogerConfig {
+	g := CatalogerConfig{
+		SearchLocalLicenses:  false,
+		SearchRemoteLicenses: false,
+		Providers:            []string{},
+		ProviderCredentials:  []nugetProviderCredential{},
+	}
+
+	nuget := os.Getenv("NUGET_SEARCH_LOCAL_LICENSES")
+	if value, err := strconv.ParseBool(nuget); err == nil {
+		g = g.WithSearchLocalLicenses(value)
+	}
+
+	remote := os.Getenv("NUGET_SEARCH_REMOTE_LICENSES")
+	if value, err := strconv.ParseBool(remote); err == nil {
+		g = g.WithSearchRemoteLicenses(value)
+	}
+
+	// process the Nuget package repository settings (for remote search)
+	nugetProviders := os.Getenv("NUGET_PACKAGE_PROVIDERS")
+	if nugetProviders == "" {
+		nugetProviders = getDefaultProviders()
+	}
+	g = g.WithProviders(nugetProviders)
+
+	// process the Nuget package repository credential settings (for remote search)
+	g = g.WithCredentials(os.Getenv("NUGET_PACKAGE_PROVIDER_CREDENTIALS"))
+
+	return g
+}
+
+func (g CatalogerConfig) WithSearchLocalLicenses(input bool) CatalogerConfig {
+	g.SearchLocalLicenses = input
+	return g
+}
+
+func (g CatalogerConfig) WithSearchRemoteLicenses(input bool) CatalogerConfig {
+	g.SearchRemoteLicenses = input
+	return g
+}
+
+func (g CatalogerConfig) WithProviders(input string) CatalogerConfig {
+	if input == "" {
+		return g
+	}
+	g.Providers = strings.Split(input, ",")
+	return g
+}
+
+func (g CatalogerConfig) WithCredentials(input string) CatalogerConfig {
+	if input == "" {
+		return g
+	}
+	g.ProviderCredentials = []nugetProviderCredential{}
+	credentials := strings.Split(input, ",")
+	for _, credential := range credentials {
+		if credentialParts := strings.Split(credential, ":"); len(credentialParts) == 2 {
+			g.ProviderCredentials = append(g.ProviderCredentials, nugetProviderCredential{
+				Username: credentialParts[0],
+				Password: credentialParts[1],
+			})
+		}
+	}
+	return g
+}
+
+type sourceAPIResource struct {
+	ID      string `json:"@id"`
```

Comment:
is this JSON-LD format?

Reply:
I referenced the JSON tags against the output of [https://api.nuget.org/v3/index.json](https://api.nuget.org/v3/index.json)...

```
{
    version: "3.0.0",
    resources: [
    {
        @id: "https://azuresearch-usnc.nuget.org/query",
        @type: "SearchQueryService",
        comment: "Query endpoint of NuGet Search service (primary)"
    },
    [...]
}
```